app/models/request.rb

```diff
@@ -27,4 +28,32 @@ def items_hash
       [key, Item.order("created_at ASC").find_by(partner_key: key)]
     end.to_h
   end
+
+  def family_request_reply
+    {
+      "organization_id": organization_id,
+      "partner_id": partner_id,
+      "requested_items": item_requests.map do |ir|
+        {
+          "item_id": ir.item_id,
+          "count": ir.quantity
+        }
+      end
+    }
+  end
+
+  def self.parse_family_request(family_request)
+    request = Request.new(organization_id: family_request['organization_id'], partner_id: family_request['partner_id'])
+    request.item_requests = []
+
+    family_request['requested_items'].each do |item|
+      item_request = ItemRequest.new
+      item_request.item = Item.find(item['item_id'])
```

Comment:
@armahillo Should I rewrite this to fetch all the items at once?

Reply:
That's really up to you. The request would be pretty small and both servers are co-located. But if you don't think it's necessary for this feature right now, we can always do it later.